csrc/transformer/transform_kernels.cu

```diff
@@ -123,10 +130,13 @@ void launch_transform_0213<float>(float* output,
                                   int heads,
                                   cudaStream_t stream)
 {
-    dim3 block_dim(hidden_dim / heads / 4, heads);
-    dim3 grid_dim(batch_size, seq_length);
-    transform_0213<float>
-        <<<grid_dim, block_dim, 0, stream>>>(output, vals, hidden_dim, seq_length, heads);
+    hidden_dim >>= 2;
+    int head_ext = (hidden_dim - 1) / 1024;

```

Comment:
Can you replace constant 1024 with a variable?

Reply:
Of course, this is the maximum number of threads allowed at each block.